onnxruntime/python/tools/kernel_explorer/kernels/gemm_fast_gelu_test.py

```diff
@@ -0,0 +1,207 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import re
+import sys
+from itertools import product
+
+import kernel_explorer as ke
+import numpy as np
+import pytest
+
+
+def transab_to_suffix(transab):
+    return {
+        (True, True): "TT",
+        (True, False): "TN",
+        (False, True): "NT",
+        (False, False): "NN",
+    }[tuple(transab)]
+
+
+def dtype_to_funcs(dtype):
+    type_map = {
+        "float16": list(filter(lambda x: re.search("GemmFastGelu.*_half", x), dir(ke))),
+        "float32": list(filter(lambda x: re.search("GemmFastGelu.*_float", x), dir(ke))),
+    }
+    return type_map[dtype]
+
+
+def fast_gelu(x, bias):
+    x = x + bias
+    y = 0.5 * x * (1 + np.tanh(0.797885 * x + 0.035677 * x * x * x))
+    return y
+
+
+# TODO The test method needs update.
+def _test_gemmfastgelu(func, dtype: str, m: int, n: int, k: int, transa=False, transb=False):
+    assert dtype in ["float16", "float32"]
+
+    a_shape = (k, m) if transa else (m, k)
+    b_shape = (n, k) if transb else (k, n)
+
+    np.random.seed(0)
+    a = (np.random.rand(*a_shape) - 0.5).astype(dtype).astype("float64")
+    b = (np.random.rand(*b_shape) + 0.5).astype(dtype).astype("float64")
+    bias = np.random.rand(n).astype(dtype)
+    temp_c = (a.T if transa else a) @ (b.T if transb else b)
+    ref_c = fast_gelu(temp_c, bias)
+
+    # The machine epsilon, unit roundoff, the smallest positive floating point number n such that the floating point
+    # number that represents 1 + n is greater than 1.
+    machine_eps = 2.0 ** -(24 if dtype == "float32" else 11)
+
+    # The following implements error bound 5.7 in paper I. C. Ipsen and H. Zhou, “Probabilistic error analysis for
+    # Inner Products,” SIAM Journal on Matrix Analysis and Applications, vol. 41, no. 4, pp. 1726–1741, 2020.
+    # NOTE: the bound is not tight for float16 when k is large
+    absa_mul_absb = np.abs(a.T if transa else a) @ np.abs(b.T if transb else b)
+    coeff = np.max(absa_mul_absb / np.abs(temp_c))
+    gamma_2k = (1.0 + machine_eps) ** (2 * k) - 1.0
+    bound_5_7 = coeff * np.sqrt(np.log(2 / 1e-10) * machine_eps * gamma_2k / 2)
+    bound = bound_5_7
```

Comment:
Could you refactor this portion and along with the copied part in `gemm_test.py`. Factorize them into a shared py file, so that we don't duplicate code. We may also switch to some tighter bound in the future.